26_Autoexposure/app_resources/luma_meter.comp.hlsl

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2024-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/luma_meter/luma_meter.hlsl"
+#include "nbl/builtin/hlsl/bda/bda_accessor.hlsl"
+#include "app_resources/common.hlsl"
+
+[[vk::combinedImageSampler]] [[vk::binding(0, 0)]] Texture2D texture;
+[[vk::combinedImageSampler]] [[vk::binding(0, 0)]] SamplerState samplerState;
+
+[[vk::push_constant]] AutoexposurePushData pushData;
+
+using Ptr = nbl::hlsl::bda::__ptr < uint32_t >;
+using PtrAccessor = nbl::hlsl::BdaAccessor < uint32_t >;
+
+groupshared float32_t sdata[WorkgroupSize];
+
+struct SharedAccessor
+{
+    uint32_t get(const uint32_t index)
+    {
+        return sdata[index];
+    }
+
+    void set(const uint32_t index, const uint32_t value)
+    {
+        sdata[index] = value;
+    }
+
+    void workgroupExecutionAndMemoryBarrier()
+    {
+        nbl::hlsl::glsl::barrier();
+    }
+};
+
+struct TexAccessor
+{
+    float32_t3 get(float32_t2 uv) {
+        return texture.Sample(samplerState, uv).rgb;
+    }
+};
+
+uint32_t3 nbl::hlsl::glsl::gl_WorkGroupSize()
+{
+    return uint32_t3(WorkgroupSize, 1, 1);
+}
+
+[numthreads(DeviceSubgroupSize, DeviceSubgroupSize, 1)]
+void main(uint32_t3 ID : SV_GroupThreadID, uint32_t3 GroupID : SV_GroupID)
+{
+    nbl::hlsl::luma_meter::LumaMeteringWindow luma_meter_window;
+    luma_meter_window.meteringWindowScale = float32_t2(pushData.meteringWindowScaleX, pushData.meteringWindowScaleY);
+    luma_meter_window.meteringWindowOffset = float32_t2(pushData.meteringWindowOffsetX, pushData.meteringWindowOffsetY);
```

Comment:
you should be using `nbl::hlsl::luma_meter::LumaMeteringWindow` directly in your pushData, instead of having to reconstitute it from individual scalars